File: packages/frontend/src/api/admin.api.ts
import api from './client';
import type { ApiResponse, AdminStats, AdminUser, Order, AdminCategory, CategoryAttribute } from '@/types';

export const adminApi = {
  getUsers: (role?: string) =>
    api.get<ApiResponse<AdminUser[]>>('/admin/users', { params: role ? { role } : {} }).then((r) => r.data.data),

  toggleUserStatus: (id: string, isActive: boolean) =>
    api.put<ApiResponse<AdminUser>>(`/admin/users/${id}`, { isActive }).then((r) => r.data.data),

  getAllOrders: () =>
    api.get<ApiResponse<Order[]>>('/admin/orders').then((r) => r.data.data),

  getStats: () =>
    api.get<ApiResponse<AdminStats>>('/admin/stats').then((r) => r.data.data),

  getCategories: () =>
    api.get<ApiResponse<AdminCategory[]>>('/admin/categories').then((r) => r.data.data),

  createCategory: (name: string, attributes?: { name: string; sortOrder?: number }[]) =>
    api.post<ApiResponse<AdminCategory>>('/admin/categories', { name, attributes }).then((r) => r.data.data),

  updateCategory: (id: string, name: string) =>
    api.put<ApiResponse<AdminCategory>>(`/admin/categories/${id}`, { name }).then((r) => r.data.data),

  deleteCategory: (id: string) =>
    api.delete<ApiResponse<null>>(`/admin/categories/${id}`).then((r) => r.data),

  createCategoryAttribute: (categoryId: string, name: string, sortOrder?: number) =>
    api
      .post<ApiResponse<CategoryAttribute>>(`/admin/categories/${categoryId}/attributes`, { name, sortOrder })
      .then((r) => r.data.data),

  updateCategoryAttribute: (
    categoryId: string,
    attributeId: string,
    data: { name?: string; sortOrder?: number },
  ) =>
    api
      .put<ApiResponse<CategoryAttribute>>(
        `/admin/categories/${categoryId}/attributes/${attributeId}`,
        data,
      )
      .then((r) => r.data.data),

  deleteCategoryAttribute: (categoryId: string, attributeId: string) =>
    api.delete<ApiResponse<null>>(`/admin/categories/${categoryId}/attributes/${attributeId}`).then((r) => r.data),
};

File: packages/mobile/lib/core/api/admin_api.dart
import 'package:dio/dio.dart';

import '../network/api_response.dart';

class AdminApi {
  AdminApi(this._dio);
  final Dio _dio;

  Future<List<dynamic>> getUsers({String? role}) async {
    final res = await _dio.get<Map<String, dynamic>>(
      '/admin/users',
      queryParameters: role != null ? {'role': role} : null,
    );
    return ApiResponse.unwrapData(res.data!, (d) => d as List<dynamic>);
  }

  Future<Map<String, dynamic>> toggleUserStatus(String id, bool isActive) async {
    final res = await _dio.put<Map<String, dynamic>>('/admin/users/$id', data: {'isActive': isActive});
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<List<dynamic>> getAllOrders() async {
    final res = await _dio.get<Map<String, dynamic>>('/admin/orders');
    return ApiResponse.unwrapData(res.data!, (d) => d as List<dynamic>);
  }

  Future<Map<String, dynamic>> getStats() async {
    final res = await _dio.get<Map<String, dynamic>>('/admin/stats');
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<List<dynamic>> getCategories() async {
    final res = await _dio.get<Map<String, dynamic>>('/admin/categories');
    return ApiResponse.unwrapData(res.data!, (d) => d as List<dynamic>);
  }

  Future<Map<String, dynamic>> createCategory(String name, List<Map<String, dynamic>>? attributes) async {
    final res = await _dio.post<Map<String, dynamic>>(
      '/admin/categories',
      data: {'name': name, if (attributes != null) 'attributes': attributes},
    );
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<Map<String, dynamic>> updateCategory(String id, String name) async {
    final res = await _dio.put<Map<String, dynamic>>('/admin/categories/$id', data: {'name': name});
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<void> deleteCategory(String id) async {
    await _dio.delete<Map<String, dynamic>>('/admin/categories/$id');
  }

  Future<Map<String, dynamic>> createCategoryAttribute(
    String categoryId,
    Map<String, dynamic> body,
  ) async {
    final res = await _dio.post<Map<String, dynamic>>(
      '/admin/categories/$categoryId/attributes',
      data: body,
    );
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<Map<String, dynamic>> updateCategoryAttribute(
    String categoryId,
    String attributeId,
    Map<String, dynamic> data,
  ) async {
    final res = await _dio.put<Map<String, dynamic>>(
      '/admin/categories/$categoryId/attributes/$attributeId',
      data: data,
    );
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<void> deleteCategoryAttribute(String categoryId, String attributeId) async {
    await _dio.delete<Map<String, dynamic>>('/admin/categories/$categoryId/attributes/$attributeId');
  }
}
